cpp/src/parquet/stream_writer.h

```diff
@@ -0,0 +1,135 @@
+#ifndef PARQUET_STREAM_WRITER_H
+#define PARQUET_STREAM_WRITER_H
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parquet/api/writer.h"
+
+namespace parquet {
+
+// A class for writing Parquet files using an output stream type API.
```

Comment:
I think doxygen commenting is preferred.

Reply:
I do not see any doxygen for the existing classes in Parquet.

Reply:
That the current codebase has problems shouldn't stop us from doing the right thing with new code =) Someone needs to make a sweep over the public API and make everything Doxygen but it is not done yet